src/btreemap.rs

```diff
@@ -217,8 +191,39 @@ where
         btree
     }
 
+    /// Create a v1 instance of the BTree.
+    ///
+    /// This is primarily exposed for testing and benchmarking.
+    pub fn new_v1(memory: M) -> Self {
+        let max_key_size = max_size::<K>();
+        let max_value_size = max_size::<V>();
+
+        let btree = Self {
+            root_addr: NULL,
+            allocator: Allocator::new(
+                memory,
+                Address::from(ALLOCATOR_OFFSET as u64),
+                Node::<K>::max_size(max_key_size, max_value_size),
+            ),
+            version: Version::V1(DerivedPageSize {
+                max_key_size,
+                max_value_size,
+            }),
+            length: 0,
+            _phantom: PhantomData,
+        };
+
+        btree.save();
+        btree
+    }
+
     /// Loads the map from memory.
     pub fn load(memory: M) -> Self {
+        Self::load_helper(memory, true)
+    }
+
+    // Loads the map from memory, potentially migrating the map from V1 to V2.
+    fn load_helper(memory: M, migrate_to_v2: bool) -> Self {
```

Comment:
Question: Would it make sense to have the migration logic guarded by a feature flag only enabled in production builds? The main advantage I see of that approach is that we can drop the boolean arg and have a single load() function.

Reply:
It's actually useful to have it as a boolean, because test-cases then have the flexibility to choose which behavior they want (whether to upgrade or not).